python/treelite/gtil/gtil.py

```diff
@@ -2,16 +2,50 @@
 General Tree Inference Library (GTIL)
 """
 import ctypes
+import json
+import warnings
+from typing import Optional
+
 import numpy as np
-from ..frontend import Model
+
 from ..core import _LIB, _check_call
+from ..frontend import Model
+from ..util import c_str
+
 
+class GTILConfig:
+    """Object holding configuration data"""
 
-def predict(model: Model, data: np.ndarray, nthread: int = -1, pred_margin: bool = False):
+    # pylint: disable=too-few-public-methods
+
+    def __init__(self, *, nthread: int, predict_type: str):
+        predictor_config = {
+            "nthread": nthread,
+            "predict_type": predict_type,
+        }
+        predictor_config = json.dumps(predictor_config)
+        self.handle = ctypes.c_void_p()
+        _check_call(
+            _LIB.TreeliteGTILParseConfig(
+                c_str(predictor_config), ctypes.byref(self.handle)
+            )
+        )
```

Comment:
I would recommend to implement this class as dataclass to avoid some of the boiler-plate code:
```suggestion
@dataclass
class GTILConfig:
    """Object holding configuration data"""

    predict_type: str
    nthread: int = -1

    def to_json(self):
        return json.dumps(asdict(self))

    def __post_init__(self):
        self.handle = ctypes.c_void_p()
        _check_call(
            _LIB.TreeliteGTILParseConfig(
                c_str(self.to_json()), ctypes.byref(self.handle)
            )
        )
```

Reply:
No idea what went wrong with the diff here...

Reply:
I like how the boilerplate is massively reduced. Thanks!